doc/user-manual/language/coinduction.lagda.rst

```diff
@@ -12,85 +15,160 @@ Coinduction
+     constructor _,_
+     field
+       fst : A
+       snd : B
+
+
+We can as well define bisimilarity (equivalence) of a pair of ``Stream A`` as a
+coinductive record.
 
-To use coinduction it is recommended that you import the module Coinduction from the `standard library <http://wiki.portal.chalmers.se/agda/pmwiki.php?n=Libraries.StandardLibrary>`_. Coinductive types can then be defined by labelling coinductive occurrences using the delay operator ``∞``:
 ::
 
-  data Coℕ : Set where
-     zero : Coℕ
-     suc  : ∞ Coℕ → Coℕ
+   record _≈_ {A : Set} (xs : Stream A) (ys : Stream A) : Set where
+     coinductive
+     field
+       hd-≈ : hd xs ≡ hd ys
+       tl-≈ : tl xs ≈ tl ys
 
-The type ``∞ A`` can be seen as a suspended computation of type ``A``. It comes with delay and force functions:
+Using copatterns we can define a pair of functions on ``Stream`` such that one
+returns a ``Stream`` with the elements in the even positions and the other the
+elements in odd positions.
+
+::
 
-.. code-block:: agda
+   even : ∀ {A} → Stream A → Stream A
+   hd (even x) = hd x
+   tl (even x) = even (tl (tl x))
 
-  ♯_ : ∀ {a} {A : Set a} → A → ∞ A
-  ♭  : ∀ {a} {A : Set a} → ∞ A → A
+   odd : ∀ {A} → Stream A → Stream A
+   odd x = even (tl x)
+
+   split : ∀ {A } → Stream A → Stream A × Stream A
+   split xs = even xs , odd xs
+
+And merge a pair of ``Stream`` by interleaving their elements.
 
-Values of coinductive types can be constructed using corecursion, which does not need to terminate, but has to be productive. As an approximation to productivity the termination checker requires that corecursive definitions are guarded by coinductive constructors. As an example the infinite “natural number” can be defined as follows:
 ::
 
-  inf : Coℕ
-  inf = suc (♯ inf)
+   merge : ∀ {A} → Stream A × Stream A → Stream A
+   hd (merge (fst , snd)) = hd fst
+   tl (merge (fst , snd)) = merge (snd , tl fst)
+
+Finally, we can prove that split is the left inverse of merge.
 
-The check for guarded corecursion is integrated with the check for size-change termination, thus allowing interesting combinations of inductive and coinductive types. We can for instance define the type of stream processors, along with some functions:
 ::
 
-  -- Infinite streams.
+   merge-split-id : ∀ {A} (xs : Stream A) → merge (split xs) ≈ xs
+   hd-≈ (merge-split-id _)  = refl
+   tl-≈ (merge-split-id xs) = merge-split-id (tl xs)
 
-  data Stream (A : Set) : Set where
-    _∷_ : (x : A) (xs : ∞ (Stream A)) → Stream A
 
-  -- A stream processor SP A B consumes elements of A and produces
-  -- elements of B. It can only consume a finite number of A’s before
-  -- producing a B.
 
-  data SP (A B : Set) : Set where
-    get : (f : A → SP A B) → SP A B
-    put : (b : B) (sp : ∞ (SP A B)) → SP A B
+.. note::
+   This is a stub.
 
-  -- The function eat is defined by an outer corecursion into Stream B
-  -- and an inner recursion on SP A B.
+-- Old Coinduction
+-- ---------------
```

Comment:
I would keep it, specially because there are still examples in some papers and on the internet where they still use this notation.

The documentation is type-checked as part of the test suite anyway, so if at some point this stops being supported by Agda, it can be quickly removed.